skimage/transform/_warps.py

```diff
@@ -1093,3 +1130,101 @@ def warp_polar(image, center=None, *, radius=None, output_shape=None,
                   output_shape=output_shape, **kwargs)
 
     return warped
+
+
+def _local_mean_weights(old_size, new_size, grid_mode, dtype):
+    """Create a 2D weight matrix for resizing with the local mean.
+
+    Parameters
+    ----------
+    old_size: int
+        Old size.
+    new_size: int
+        New size.
+    grid_mode : bool
+        Whether to use grid data model of pixel/voxel model for
+        average weights computation.
+    dtype: dtype
+        Output array data type.
+
+    Returns
+    -------
+    weights: (new_size, old_size) array
+        Rows sum to 1.
+
+    """
+    if grid_mode:
+        old_breaks = np.linspace(0, old_size, num=old_size + 1, dtype=dtype)
+        new_breaks = np.linspace(0, old_size, num=new_size + 1, dtype=dtype)
+    else:
+        old, new = old_size - 1, new_size - 1
+        old_breaks = np.pad(np.linspace(0.5, old - 0.5, old, dtype=dtype),
+                            1, 'constant', constant_values=(0, old))
+        if new == 0:
+            val = np.inf
+        else:
+            val = 0.5 * old / new
+        new_breaks = np.pad(np.linspace(val, old - val, new, dtype=dtype),
+                            1, 'constant', constant_values=(0, old))
+
+    upper = np.minimum(new_breaks[1:, np.newaxis], old_breaks[np.newaxis, 1:])
+    lower = np.maximum(new_breaks[:-1, np.newaxis],
+                       old_breaks[np.newaxis, :-1])
+
+    weights = np.maximum(upper - lower, 0)
+    weights /= weights.sum(axis=1, keepdims=True)
+
+    return weights
+
+
+def resize_local_mean(image, output_shape, grid_mode=True,
+                      preserve_range=False):
+    """Resize an array with the local mean / bilinear scaling.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    output_shape : tuple or ndarray
+        Size of the generated output image `(rows, cols[, ...][, dim])`. If
+        `dim` is not provided, the number of channels is preserved. In case the
+        number of input channels does not equal the number of output channels a
+        n-dimensional interpolation is applied.
+    grid_mode : bool, optional
+        Whether to use grid data model of pixel/voxel model for
+        average weights computation.
```

Comment:
```suggestion
        If False, the distance from the pixel centers is zoomed. Otherwise, the
        distance including the full pixel extent is used. For example, a 1d
        signal of length 5 is considered to have length 4 when `grid_mode` is
        False, but length 5 when `grid_mode` is True. See the following
        visual illustration:

        .. code-block:: text

                | pixel 1 | pixel 2 | pixel 3 | pixel 4 | pixel 5 |
                     |<-------------------------------------->|
                                        vs.
                |<----------------------------------------------->|

        The starting point of the arrow in the diagram above corresponds to
        coordinate location 0 in each mode.
```

This is a copy of a docstring I wrote for the parameter of the same that was introduced to `scipy.ndimage.zoom` in SciPy 1.6. Hopefully it helps clarify the difference between the two modes